habitat/core/dataset.py

```diff
@@ -376,9 +377,6 @@ def _switch_scene(self) -> None:
 
         if len(grouped_episodes) > 1:
             # Ensure we swap by moving the current group to the end
-            if self.shuffle:
```

Comment:
Because, shuffles are already happening at the beginning and on each cycle. That's why this shuffle is redundant and can be omitted with no test failing.

Reply:
Yeah, I guess both logics are valid.  Either you cycle through all scenes in a predetermined but random order, or you randomly draw a new scene from the set of other scenes.  One doesn't seem inherently better than the other.

Reply:
My point is that it's already in `predetermined but random order` because of `self.shuffle` in other places. So there is no need to shuffle again.

Reply:
Yep yep, it's just different ideas of how scenes should be selected on a swap.  I am fine with this way if it makes more sense to you, both make sense to me.